scipy/signal/tests/test_signaltools.py

```diff
@@ -2654,3 +2655,17 @@ def test_basic(self):
         recorded = [0, 2, 1, 0, 2, 3, 1, 0, 0]
         recovered, remainder = signal.deconvolve(recorded, impulse_response)
         assert_allclose(recovered, original)
+
+
+class TestDetrend(object):
+
+    def test_basic(self):
+        detremded = detrend(array([1, 2, 3]))
+        detrended_exact = array([0, 0, 0])
+        assert_array_almost_equal(detremded, detrended_exact)
+
+    def test_copy(self):
+        x = np.ones(10)
+        overwritten = detrend(x, overwrite_data=True)
```

Comment:
this test is incorrect, and only happens to pass because `x` has slope zero. you need to do the non-inplace detrend first. can you change `x` so this fails, and then swap the order of the detrends?

Reply:
I've added some semi-random numbers to give a slope not equal to zero, and swapped the order around. I also fixed a typo I made in the test done previous, and reverted the spacing changes pycharm made when.